File: rulepacks/2026.08.2/testdata/datos-ext.ts
// Fixture de semgrep --test — no es código del producto.
// Casos de prueba del pilar datos (stem datos-ext), lenguaje TypeScript.

// --- pii-en-telemetria ---------------------------------------------------------------
export function rastrearAlta(correo: string, plan: string) {
  // ruleid: pii-en-telemetria
  analytics.track("alta de cuenta", { email: correo });
  // ok: pii-en-telemetria
  analytics.track("alta de cuenta", { plan: plan });
}

// --- ts-datetime-naive ---------------------------------------------------------------
export function crearFechas() {
  // ruleid: ts-datetime-naive
  const inicio = new Date("2026-01-15");
  // ok: ts-datetime-naive
  const fin = new Date("2026-01-15T00:00:00-06:00");
  return [inicio, fin];
}

// --- escrituras-sin-transaccion-ts ------------------------------------------------------
export async function guardarPedido(db: any, pedido: any, reserva: any) {
  // ruleid: escrituras-sin-transaccion-ts
  db.update(pedido);
  db.delete(reserva);
}

export async function guardarPedidoAtomico(sequelize: any, pedido: any, reserva: any) {
  await sequelize.transaction(async (tx: any) => {
    // ok: escrituras-sin-transaccion-ts
    tx.update(pedido);
    tx.delete(reserva);
  });
}

// --- ts-dinero-float -----------------------------------------------------------------
// ruleid: ts-dinero-float
const precioBase: number = 99.9;
// ruleid: ts-dinero-float
let saldoPendiente: number = 0;

// ruleid: ts-dinero-float
export function cobrar(importeFinal: number) {
  return importeFinal;
}

// El nombre no es monetario: la regla filtra por nombre.
// ok: ts-dinero-float
const cantidadItems: number = 3;
// Dinero como entero en centavos, el remedio recomendado.
// ok: ts-dinero-float
const precioCentavos: bigint = 9990n;

File: rulepacks/2026.08.2/testdata/datos-ext.py
# Fixture de semgrep --test — no es código del producto.
# Casos de prueba del pilar datos (stem datos-ext), lenguaje Python.
from datetime import datetime, timezone


# --- fixtures-datos-reales (regla generic, se prueba en este archivo) ----------
# ruleid: fixtures-datos-reales
CORREO_SEMILLA = "laura.mtz@dominioreal.com"
# ok: fixtures-datos-reales
CORREO_PRUEBA = "laura.mtz@example.com"


# --- python-dinero-float --------------------------------------------------------
def convertir_importes(fila):
    # ruleid: python-dinero-float
    precio_unitario = float(fila["precio"])
    # ok: python-dinero-float
    latitud = float(fila["latitud"])
    return precio_unitario, latitud


# --- python-datetime-naive ------------------------------------------------------
def marcar_tiempos():
    # ruleid: python-datetime-naive
    creado = datetime.now()
    # ok: python-datetime-naive
    actualizado = datetime.now(timezone.utc)
    return creado, actualizado


# --- log-dato-sensible ------------------------------------------------------------
def registrar_acceso(logger, usuario, password):
    # ruleid: log-dato-sensible
    logger.info("intento fallido", password)
    # ok: log-dato-sensible
    logger.info("acceso correcto", usuario.id)


# --- pii-en-telemetria-py ---------------------------------------------------------
def identificar_sesion(sentry_sdk, analytics, correo, plan):
    # ruleid: pii-en-telemetria-py
    sentry_sdk.set_user({"email": correo})
    # ruleid: pii-en-telemetria-py
    analytics.track("alta", {"telefono_contacto": correo})
    # ok: pii-en-telemetria-py
    analytics.track("alta", {"plan": plan})


# --- pii-en-url ------------------------------------------------------------------------
def construir_urls(correo, uid):
    # ruleid: pii-en-url
    busqueda = f"https://interno.example/usuarios?email={correo}"
    # ok: pii-en-url
    ficha = f"https://interno.example/usuarios?id={uid}"
    return busqueda, ficha


# --- escrituras-sin-transaccion -----------------------------------------------------
def guardar_pedido(repo, pedido, reserva):
    # ruleid: escrituras-sin-transaccion
    repo.save(pedido)
    repo.delete(reserva)


def guardar_pedido_atomico(store, transaction, pedido, reserva):
    with transaction.atomic():
        # ok: escrituras-sin-transaccion
        store.save(pedido)
        store.delete(reserva)


# --- sql-truncate-o-drop -----------------------------------------------------------
def limpiar_sesiones(cursor):
    # ruleid: sql-truncate-o-drop
    cursor.execute("TRUNCATE TABLE sesiones_caducadas")
    # ok: sql-truncate-o-drop
    cursor.execute("DELETE FROM sesiones_caducadas WHERE edad_dias > 30")
